### DISCUSSION TITLE
[Icon][Linux] Semi-Finale Poll

### DISCUSSION THREAD

**Discussion by daiyam** [VSCodium/vscodium#1488] [Polls]
The Semi-Finale will ends the 25th April at midnight.
The Finale will start on the 25th with the 2 finalists and will end the 1st May at midnight.

Most taskbars will use the small icon (32px), click on the image to see it at full size.
The second line is a blue overlay used by some DE, to indicate that the app is active.

| icon | preview |
|-|-|
| `circle` |![Screen Shot 2023-04-19 at 12 55 27 AM](https://user-images.githubusercontent.com/587742/232922443-455b90a6-6cdc-4df3-bd95-e1b3c0219618.png)| 
| `nobg` | ![Screen Shot 2023-04-19 at 12 55 45 AM](https://user-images.githubusercontent.com/587742/232922497-90548333-5f39-4ead-9259-cdf5ea51dee8.png) |
| `border` | ![Screen Shot 2023-04-19 at 12 56 01 AM](https://user-images.githubusercontent.com/587742/232922537-e0cfab98-3af1-46f5-8d50-123dde5d1015.png) |

**Comment by DanielProg39** [comment]
In case of border, I think that the border itself should be white, because it will make the icon more distinguishable if the background is too dark or is the same as the icon's color

**Reply by geekley** [reply]
See [my post below](#discussioncomment-5696970) for why it was made this way (tl;dr to solve `nobg` contrast issue on Unity DE without deviating too much from its look, specially at small sizes). An option could be lighter blue instead of full white.
I came up with this, but tbh I prefer `circle` for how it looks next to other apps.

**Comment by mndscp** [comment]
Thanks for pinging in the other thread @daiyam, and for all the effort you're putting into this and for listening to opinions.
I love the circle one. This discussion has made me look at system icons with new eyes and most of the icons that don't have any border or contrasting background have the same issues (KDE on Linux here). Another thank you to everyone else here who voiced their opinions :)

From a design perspective the nobg doesn't stand out enough and drowns in backgrounds as others have mentioned and the one with the border gets really messy the smaller the icon. Circle looks like a proper application icon and very clean in my opinion.

I've been using the SVG from [this post](https://github.com/VSCodium/vscodium/discussions/1218#discussioncomment-5586517) and it looks great to me. It does have that tiny transparent bottom border. Can't even notice that in the application menu as you can see in the screenshot but I do question its necessity. I'm so glad that circle seems to be winning. 

Is it the same SVG you are going to use if circle wins?

![image](https://user-images.githubusercontent.com/81937645/233108371-6e68ef85-85a8-47a8-abf2-388c94b83445.png)

**Reply by geekley** [reply]
If think if `circle` wins by a large difference (at least absolute majority) and if there's desire to do a finale anyway, then it could be about what shadow to put in circle. Shadow options could be:
- `flat` (this version; which someone called "pseudo-3D")
- `smooth` (my SVG from the post you linked, maybe with a slightly less dark shadow color)
- `none` (centralize circle with no shadow or border at all)
- `stroke` (centralize and put a small blackish border around the white circle)

Though tbh I personally think this `flat` version here is already great; it being gray bothered me a lot less than I thought it would at 32px, and I guess kinda works for both light and dark themes.
I'd be fine with any of these options but suggesting because, you know, detailsssss, detailsssss :laughing:

**Comment by evur** [comment]
Please remove the shadows, if they are not just for previewing.

**Comment by masakk1** [comment]
The border one is cool but... Why is the border so think? A thinner outline might be pretty cool

**Reply by mndscp** [reply]
And actually why is the border blue? Shouldn't the purpose of the border be to make the icon more visible on dark/blue backgrounds? This way it kind of extends the icon but doesn't do much about the contrast. I had expected a thinner white outline.
Still voting for the circle version, don't get me wrong :)

**Reply by geekley** [reply]
> And actually why is the border blue? Why is the border so thick?

Look at [this post](https://github.com/VSCodium/vscodium/discussions/1477#discussioncomment-5620417) and the one below it. I made it thinner than before, but just enough so it still works at 16px.
Previews below (of previous versions).
<img src="https://raw.githubusercontent.com/VSCodium/vscodium/99da974ce5f1126a74c8bfedc07e3103a61c5272/icons/stable/codium_cnl_w80_b8.svg" height=16/>
<img src="https://user-images.githubusercontent.com/38332912/232164407-98fa1978-0a8c-4f67-947a-1bc0de98f0d8.png" height=16/> <img src="https://user-images.githubusercontent.com/38332912/232164401-6cd7fa92-f133-474a-9c0a-eafd95523f57.png" height=16/>

Full thick white border has been attempted, but looks bad at small sizes. My intention was to offer an alternative to `nobg` that would solve the contrast issue ***while trying not to deviate too much from the `nobg` aesthetic*** (since it was winning that poll), so it would hopefully not have a lot of rejection in case `circle`/`rect`/`macos` didn't win and it was picked as an alternative to `nobg` without the contrast issue. I tried many different white border sizes, and opacity values to find something that would look acceptable at 16px until I remembered I could just use a darker variation of the same blue.

The contrast issue is somewhat bad on KDE dark theme (colors on this poll's previews above), but **the REAL issue is on Unity DE**, which uses the **exact same** icon color on purpose. So border was *just enough* to solve it.
![](https://user-images.githubusercontent.com/5273753/232200955-38eab0fe-adc5-4a31-98e5-710fca3569ee.png) ![](https://user-images.githubusercontent.com/5273753/232201229-e6b02eac-ba82-45e2-9e1d-cb52be183dca.png)

As for why "darker" instead of "lighter", that was a mere arbitrarity; I added this comment as an afterthought:
> I went with a-bit-darker-blue stroke, but it might also work with a-bit-lighter-blue stroke.

So, light blue might work. White won't work for this kind of icon. Tbh I already don't even like `border` (even though I came up with its idea) because, while it's fine on its own, it's too different next to other app icons. I'm rooting for `circle`, which just works.

**Comment by MrDuartePT** [comment]
How about turn the circle black on dark mode this would be cool

**Reply by evur** [reply]
Interesting idea. Though I think it should be the inverse (white circle on a dark panel/dock/background) :)

**Reply by MrDuartePT** [reply]
On second though I agree would be better.

**Reply by daiyam** [reply]
Feel free to make your variants available to all at https://github.com/VSCodium/icons
https://github.com/VSCodium/icons/blob/main/CONTRIBUTING.md might help

**Comment by subcidr** [comment]
Great work, my vote goes to `circle` ✨
